### DISCUSSION TITLE
loading using hugging face `AutoModel.from_pretrained`

### DISCUSSION THREAD

**Discussion by jyotipmahes** [mlfoundations/open_clip#615] [General]
How to i save the model in such a way that it can be easily loaded using hugging face `AutoModel.from_pretrained`. I looked at the code push_to_hf_hub.py however it does not seem to create all the files such as config.json and model file with correct name that can be used to loading using HF.

**Comment by rwightman** [comment]
@jyotipmahes push to hub here is to push in OpenCLIP format to the hub. To push for transformers you have to use a model with the default builtin ViT and text transformer, and then you need to convert the checkpoints and upload a transformers config to hub ... I had a hacky script here https://gist.github.com/rwightman/c79fd0241ed3c860e898114931c07990
